crates/octez/src/node_config.rs

```diff
@@ -0,0 +1,278 @@
+#[derive(Default)]
+pub struct OctezNodeRunOptionsBuilder {
+    synchonisation_threshold: Option<u8>,
+    network: Option<String>,
+    history_mode: Option<OctezNodeHistoryMode>,
+}
+
+impl OctezNodeRunOptionsBuilder {
+    pub fn new() -> Self {
+        OctezNodeRunOptionsBuilder {
+            ..Default::default()
+        }
+    }
+
+    pub fn set_synchronisation_threshold(&mut self, threshold: u8) -> &mut Self {
+        self.synchonisation_threshold.replace(threshold);
+        self
+    }
+
+    pub fn set_network(&mut self, network: &str) -> &mut Self {
+        self.network.replace(network.to_owned());
+        self
+    }
+
+    pub fn set_history_mode(&mut self, mode: OctezNodeHistoryMode) -> &mut Self {
+        self.history_mode.replace(mode);
+        self
+    }
+
+    pub fn build(&mut self) -> OctezNodeRunOptions {
+        OctezNodeRunOptions {
+            synchonisation_threshold: self
+                .synchonisation_threshold
+                .take()
+                .unwrap_or_default(),
+            network: self.network.take().unwrap_or(DEFAULT_NETWORK.to_owned()),
+            history_mode: self.history_mode.take(),
+        }
+    }
+}
+
+#[derive(Default, Clone)]
+pub struct OctezNodeConfig {
+    /// Path to the octez node binary.
+    pub binary_path: PathBuf,
+    /// Path to the directory where the node keeps data.
+    pub data_dir: PathBuf,
+    /// Name of the tezos network that the node instance runs on.
+    pub network: String,
+    /// HTTP endpoint of the node RPC interface, e.g. 'localhost:8732'
+    pub rpc_endpoint: String,
+    /// Path to the file that keeps octez node logs.
+    pub log_file: PathBuf,
+    /// Run options for octez node.
+    pub run_options: OctezNodeRunOptions,
+}
+
+#[derive(Default)]
+pub struct OctezNodeConfigBuilder {
+    binary_path: Option<PathBuf>,
+    data_dir: Option<PathBuf>,
+    network: Option<String>,
+    rpc_endpoint: Option<String>,
+    log_file: Option<PathBuf>,
+    run_options: Option<OctezNodeRunOptions>,
+}
+
+impl OctezNodeConfigBuilder {
+    pub fn new() -> Self {
+        OctezNodeConfigBuilder::default()
+    }
+
+    /// Sets the path to the octez node binary.
+    pub fn set_binary_path(&mut self, path: &str) -> &mut Self {
+        self.binary_path = Some(PathBuf::from(path));
+        self
+    }
+
+    /// Sets the path to the directory where the node keeps data.
+    pub fn set_data_dir(&mut self, path: &str) -> &mut Self {
+        self.data_dir = Some(PathBuf::from(path));
+        self
+    }
+
+    /// Sets the name of the tezos network that the node instance runs on.
+    pub fn set_network(&mut self, network: &str) -> &mut Self {
+        self.network = Some(network.to_owned());
+        self
+    }
+
+    /// Sets the HTTP(S) endpoint of the node RPC interface, e.g. 'http://localhost:8732'
+    pub fn set_rpc_endpoint(&mut self, endpoint: &str) -> &mut Self {
+        self.rpc_endpoint = Some(endpoint.to_owned());
+        self
+    }
```

Comment:
isn't this `locahost:8732` instead of `http://localhost:8732`? imo using the `Endpoint` struct would be better to avoid such mistakes

Reply:
This line was copied from octez_node.rs and that original line is updated in #589. I've corrected this line.